zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java

```diff
@@ -1081,12 +1082,48 @@ public Builder setAlertRef(String alertRef) {
             return this;
         }
 
-        /** @since 2.11.0 */
+        /** @since 2.11.0 @Deprecated 2.12.0 Replaced by {@link #setAlertTags(Map)} */
         public Builder setTags(Map<String, String> tags) {
+            return setAlertTags(tags);
+        }
+
+        /** @since 2.12.0 */
+        public Builder setAlertTags(Map<String, String> tags) {
             this.tags = tags;
             return this;
         }
 
+        /**
+         * Adds a tag to the existing collection for this Alert/Builder.
+         *
+         * @since 2.12.0
+         */
+        public Builder addAlertTag(String tag, String value) {
+            this.tags.put(tag, value);
+            return this;
+        }
+
+        /**
+         * Removes a tag with the given key (tag/name) from the existing collection for this
+         * Alert/Builder.
+         *
+         * @since 2.12.0
+         */
+        public Builder removeAlertTag(String tag) {
+            this.tags.remove(tag);
+            return this;
+        }
+        /**
+         * Removes a tag with the given key (tag/name) and value from the existing collection for
+         * this Alert/Builder.
+         *
+         * @since 2.12.0
+         */
+        public Builder removeAlertTag(String tag, String value) {
+            this.tags.remove(tag, value);
+            return this;
+        }
```

Comment:
I figured if I was introducing an add method then it made sense to also introduce a remove method.